### PR TITLE
[GLUTEN-2682][VL] support array extract

### PR DESCRIPTION
**Pull request by summaryzb**
## What changes were proposed in this pull request?
Fixes: [ISSUE-2682](https://github.com/oap-project/gluten/issues/2682)

## How was this patch tested?
pass ut

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/expression/ArrayExpressionTransformer.scala [hunk 1]
```diff
@@ -93,10 +94,17 @@ class GetArrayItemTransformer(
...
 
-    val functionName = ConverterUtils.makeFuncName(
-      substraitExprName,
-      Seq(original.left.dataType, original.right.dataType),
-      FunctionConfig.OPT)
+    val functionName = if (BackendsApiManager.isVeloxBackend) {
+      ConverterUtils.makeFuncName(
+        ExpressionMappings.expressionsMap.get(classOf[ElementAt]).get,
+        Seq(original.dataType),
+        FunctionConfig.OPT)
```
Comment (jackylee-ch) [inline gluten-core/src/main/scala/io/glutenproject/expression/ArrayExpressionTransformer.scala:101]: Please add some uts for this implement.

Comment (summaryzb) [reply gluten-core/src/main/scala/io/glutenproject/expression/ArrayExpressionTransformer.scala:101]: ComplexTypeSuite does the ut

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala [hunk 2]
```diff
@@ -57,6 +60,45 @@ import scala.collection.mutable.ArrayBuffer
...
+        FunctionConfig.OPT))
+    // right node already add 1
+    val literalNode = ExpressionBuilder.makeLiteral(1.toInt, IntegerType, false)
+    val lessThanFuncNode = ExpressionBuilder.makeScalarFunction(
+      lessThanFuncId,
+      Lists.newArrayList(rightNode, literalNode),
+      ConverterUtils.getTypeNode(BooleanType, true))
+    new IfThenNode(Lists.newArrayList(lessThanFuncNode), Lists.newArrayList(nullNode), resultNode)
+  }
+
```
Comment (yma11) [inline backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala:101]: Why not implement Spark `element_at` at velox side? Current solution seems like a workaround at transformer layer and it hides the difference between Spark `get_array_item` and Presto `element_at`. Furthermore, we may have to do such a workaround for Map `element_at` in future.

Comment (jackylee-ch) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala:101]: Actually, the `element_at` function in Spark is consistent with the one in Velox. It is the`get_array_item` function in Spark that has some semantic differences with the `element_at` function in Spark. 
Here’s an example to illustrate the difference between `get_array_item` and `element_at` in Spark.
GetArrayItem UT:
https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ComplexTypeSuite.scala#L63
element_at UT:
https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala#L2130

Comment (yma11) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala:101]: So do we still need to do implementation of `get_array_item` at velox side? I see you don't change the function name in transformer.

Comment (jackylee-ch) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala:101]: AFAIK, it is not nessesary to support `get_array_item` in velox now. It seems that current implement has already solve the difference between `get_array_tiem` and `element_at`.

Comment (yma11) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala:101]: You resolve the difference here but still need to map `get_array_item` to `element_at` for offloading to velox, as changes in [2760](https://github.com/oap-project/gluten/pull/2760). The current status is fallback. You can confirm and I will reopen the PR if need.

Comment (summaryzb) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala:101]: After apply this pr, it'll not fallback

Comment (summaryzb) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecHandler.scala:109]: > So do we still need to do implementation of `get_array_item` at velox side? I see you don't change the function name in transformer.

function name changed in `genGetArrayItemExpressionNode`

**Comment by jackylee-ch** [comment]
cc @zhouyuan @zhztheplayer

**Review by jackylee-ch — DISMISSED** [review]

**Review by jackylee-ch — APPROVED** [review]
LGTM

**Review by ulysses-you — COMMENTED** [review]
better to add some tests

**Comment by jackylee-ch** [comment]
> better to add some tests

@ulysses-you There are already some uts in [GlutenComplexTypeSuite](https://github.com/oap-project/gluten/blob/main/gluten-ut/spark33/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenComplexTypeSuite.scala#L21).

**Pull request merged**